metals/src/main/scala/scala/meta/internal/metals/ClientConfiguration.scala

```diff
@@ -4,20 +4,33 @@ import scala.meta.internal.metals.Configs.GlobSyntaxConfig
 import scala.meta.internal.metals.config.DoctorFormat
 import scala.meta.internal.metals.config.StatusBarState
 
+import org.eclipse.lsp4j.InitializeParams
+
 /**
  * This class provides a uniform way to know how the client is configured
  * using a combination of server properties, `clientExperimentalCapabilities`
  * and `initializationOptions`.
  *
  * @param initalConfig Initial server properties
- * @param experimentalCapabilities clientExperimentalCapabilities
- * @param initializationOptions initializationOptions
  */
-class ClientConfiguration(
-    var initialConfig: MetalsServerConfig,
-    var experimentalCapabilities: ClientExperimentalCapabilities,
-    var initializationOptions: InitializationOptions
-) {
+case class ClientConfiguration(initialConfig: MetalsServerConfig) {
+
+  private var experimentalCapabilities = ClientExperimentalCapabilities.Default
+  private var initializationOptions = InitializationOptions.Default
+  private var refreshSupport = false
+
+  def update(params: InitializeParams): Unit = {
+    experimentalCapabilities =
+      ClientExperimentalCapabilities.from(params.getCapabilities)
+    initializationOptions = InitializationOptions.from(params)
+    val capabilities = params.getCapabilities()
+    val codeLenseRefreshSupport: Option[Boolean] = for {
+      workspace <- Option(capabilities.getWorkspace())
+      codeLens <- Option(workspace.getCodeLens())
+      refreshSupport <- Option(codeLens.getRefreshSupport())
+    } yield refreshSupport
+    refreshSupport = codeLenseRefreshSupport.getOrElse(false)
```

Comment:
Maybe style preference, but I don't love having the `private var refreshSupport` especially if we end up with more of these in the future.

What do you think instead having `private var capabilities`. Then have the `codeLenseRefreshSupport` method just pull it from there. That also would mean it'd need to do that each time, which isn't ideal, but it's odd to me that we have `refreshSupport` up there since it doesn't seem to fit along with the other `experimentalCababilities` and `initializationOptions` and the pattern of having the methods grab values out of them. Wdyt?